### PR TITLE
Refactor hollow-node into separate components

### LINKED ISSUE #45650: Refactor hollow-node into hollow-kubelet and hollow-proxy & make them config-driven
Currently it's a mess having the same file for both hollow- kubelet and proxy that morphs into one or the other based on a flag.
- This should be refactored into separate files for each.
- Define the right set of configs/flags for either and not take everything together.
- Move almost all logic in hollow-node.go to hollow-kubelet.go and hollow-proxy.go, so we have useful libraries for them.

cc @wojtek-t @gmarek

### PR DESCRIPTION
**Pull request by Jeffwan**
1. Refactor hollow-node into hollow-kubelet and hollow-proxy

Signed-off-by: Jiaxin Shan <seedjeffwan@gmail.com>

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Refactor hollow-node into hollow-kubelet and hollow-proxy and make config driven. 
Currently, we use `--morph` to differentiate two components. As parameter grows, it's better to split into separate components/

#### Which issue(s) this PR fixes:

Fixes #45650

#### Does this PR introduce a user-facing change?

```release-note
Remove usage `kubemark --morph=kubelet` and use `hollow-kubelet` and `hollow-proxy` instead.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cmd/kubemark/hollow-kubelet/hollow-kubelet.go [hunk 1]
```diff
@@ -28,73 +27,54 @@ import (
...
-	ProxierSyncPeriod    time.Duration
-	ProxierMinSyncPeriod time.Duration
-	NodeLabels           map[string]string
-	RegisterWithTaints   []core.Taint
+type hollowKubeletConfig struct {
+	KubeconfigPath      string
+	KubeletPort         int
+	KubeletReadOnlyPort int
+	NodeName            string
+	ServerPort          int
```
Comment (jkaniuk) [inline cmd/kubemark/hollow-kubelet/hollow-kubelet.go:54]: Is ServerPort used for anything?

File: cmd/kubemark/hollow-proxy/hollow-proxy.go [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+	if err != nil {
+		return nil, fmt.Errorf("error while creating kubeconfig: %v", err)
+	}
+	config.ContentType = c.ContentType
+	config.QPS = 10
+	config.Burst = 20
+	return config, nil
+}
+
+
```
Comment (jkaniuk) [inline cmd/kubemark/hollow-proxy/hollow-proxy.go:82]: too many line breaks

Comment (Jeffwan) [reply cmd/kubemark/hollow-proxy/hollow-proxy.go:82]: I will remove empty lines

File: cmd/kubemark/hollow-kubelet/hollow-kubelet.go (hunk 1 above)
Comment (Jeffwan) [reply cmd/kubemark/hollow-kubelet/hollow-kubelet.go:54]: I double checked this field is not being used and I can remove it.  Even in hollow-node.go version, it's not used.

**Comment by Jeffwan** [comment]
I need to double check kubemark reference in test cases and make sure kubemark-e2e pass.

**Pull request closed without merging**